src/main/java/biz/paluch/logging/gelf/intern/sender/GelfHTTPSender.java

```diff
@@ -76,7 +76,12 @@ public boolean sendMessage(GelfMessage message) {
             errorReporter.reportError("Server responded with unexpected status code: " + responseCode, null);
 
         } catch (IOException e) {
-            errorReporter.reportError("Cannot send data to " + url, e);
+            String cleanUrl = url.toString();
+            String userInfo = url.getUserInfo();
+            if (userInfo != null) {
+                cleanUrl = cleanUrl.replace(userInfo + "@", "");
```

Comment:
This call isn't safe because `replace(…)` expects a regex. If `userInfo` contains Regex control characters, then these get passed-thru.

Reply:
The replace() call doesn't treat the parameter as regex but a literal.  I added a testcase for coverage (and tried with replaceAll() just to see the difference).